starlark/library.go

```diff
@@ -1780,6 +1780,18 @@ func string_format(fnname string, recv_ Value, args Tuple, kwargs []Tuple) (Valu
 	return String(buf.String()), nil
 }
 
+func decimal(s string) (x int, ok bool) {
+	n := len(s)
+	for i := 0; i < n; i++ {
+		digit := s[i] - '0'
+		if digit > 9 {
+			return 0, false // not in [0-9]
+		}
+		x = x*10 + int(digit)
```

Comment:
```go
if x < 0 {
  return 0, false // overflow
}
```

Reply:
The final condition (0 < n && n < 10) is false if there was any possibility of underflow.

Reply:
Done.